_posts/2025-04-28-sparse-autodiff.md

```diff
@@ -398,18 +408,21 @@ But if we fully materialize a Jacobian via traditional AD, ASD isn't needed.
     </div>
 </div>
 <div class="caption">
-    Figure 10: The two elementary steps in ASD: (a) sparsity pattern detection, (b) coloring of the sparsity pattern.
+    Figure 10: The first two steps of ASD: (a) sparsity pattern detection, (b) coloring of the sparsity pattern.
 </div>
 
-The solution to this problem is shown in Figure 10:
-in order to find orthogonal columns (or rows), we don't need to materialize the full Jacobian.
-Instead, it is enough to materialize a binary sparsity pattern of the Jacobian.
-This pattern contains enough information to color it.
+To sum up, ASD consists of four steps:
 
-Performance is key: For one-off computations, these two steps need to be faster than the computation of columns or rows they allow us to skip. Otherwise, we didn't gain any performance.
-As we will see in later benchmarks, this level of performance can be achieved.
-Additionally, if we need to compute Jacobians multiple times and are able to reuse the sparsity pattern, 
-the cost of sparsity pattern detection and coloring can be amortized over time.
```

Comment:
Because these steps don't _need_ to be faster to be useful. Even if they are slower, sparse AD can still be beneficial due to pattern and coloring reuse. That's why I reformulated it slightly differently below.
I know that in our tracer paper we argue for single-use detection but here it's not the focus.